packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -10,21 +10,64 @@ import 'framework.dart';
 import 'ticker_provider.dart';
 import 'transitions.dart';
 
+// Internal representation of a child that, now or in the past, was set on the
+// AnimatedChildSwitcher.child field, but is now in the process of
+// transitioning. The internal representation includes fields that we don't want
+// to expose to the public API (like the controller).
 class _AnimatedChildSwitcherChildEntry {
   _AnimatedChildSwitcherChildEntry(this.widget, this.controller, this.animation);
 
+  // The widget that is currently, or used to be, set as the widget's child.
   Widget widget;
 
+  // The animation controller for the child's transition.
   final AnimationController controller;
+
+  // The animation describing the child's transition, with the controller as
+  // its parent.
+  final Animation<double> animation;
+}
+
+/// Represents a child that used to be set on the [AnimatedChildSwitcher.child]
+/// field, but is now in the process of transitioning out.
+class AnimatedChildSwitcherLayoutData {
+  AnimatedChildSwitcherLayoutData(this.widget, this.animation);
+
+  /// The widget that is currently or used to be set as the widget's child.
+  Widget widget;
```

Comment:
presumably this should be final?

Reply:
or at least, just a getter?

Reply:
If you do the thing above with the abstract class bla bla, you'll probably want it to be a getter in this interface, and a mutable field in the concrete subclass.